docs/run-keep-ecdsa.adoc

```diff
@@ -0,0 +1,419 @@
+:numbered:
+toc::[]
+
+== System Considerations
+
+The Keep Network expects certain capabilites for each node running on the network.  To help attain
+these capabilities consider the following criteria:
+
+- It is paramount that Keep nodes remain available to the Keep Network. We strongly encourage a
+  stable and redundant internet connection.
+- A connection to a production grade self-hosted or third party Ethereum node deployment.
+- Persistent and redundant storage that will survive a VM or container rotation, and disk failure.
+- Each Keep ECDSA client running on the network requires a unique Ethereum operator account.
+- Each Keep ECDSA client running on the network requires a unique IP address or a unique
+  application port running under the same IP.
+- Recommended machine types by provider:
+
+[%header,cols=2*]
+|===
+|Cloud Provider
+|Machine Type
+
+|Google Cloud
+|n1-standard-4
+
+|AWS
+|m5.xlarge
+
+|Azure
+|D4s v3
+
+|Self-hosted
+|4 vCPU / 4 GiB RAM / 1 GiB Persistent Storage
+|===
+
+== Configuration
+
+=== Network
+
+Default port mappings.
+
+[%header,cols=2*]
+|===
+|Egress
+|Port
+
+|Ethereum Network
+|`8545` / `8546`
+
+|Keep Network
+|`3919`
+|===
+
+[%header,cols=2*]
+|===
+|Ingress
+|Port
+
+|Keep Network
+|`3919`
+|===
+
+If you set a different `port` in your keep-ecdsa configuration, or configure `peers` with
+non-default ports configured, firewall rules will need to be adjusted accordingly.
+
+=== Application
+
+Application configurations are stored in a `.toml` file and passed to the application run command
+ with the `--config` flag.
+
+==== Sample
+
+```
+# This is a TOML configuration file.
+
+# Connection details of ethereum blockchain.
+[ethereum]
+  URL = "ws://127.0.0.1:8545"
+  URLRPC = "http://127.0.0.1:8546"
+
+[ethereum.account]
+  KeyFile = "/Users/someuser/ethereum/data/keystore/UTC--2018-03-11T01-37-33.202765887Z--AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8AAAAAAAAA"
+
+# Addresses of contracts deployed on ethereum blockchain.
+[ethereum.ContractAddresses]
+  BondedECDSAKeepFactory = "0xCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
+
+# Addresses of applications approved by the operator.
+[SanctionedApplications]
+  Addresses = [
+    "0xDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD",
+    "0xEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"
+  ]
+
+[Storage]
+  DataDir = "/my/secure/location"
+
+[LibP2P]
+  Peers = ["/ip4/127.0.0.1/tcp/3919/ipfs/njOXcNpVTweO3fmX72OTgDX9lfb1AYiiq4BN6Da1tFy9nT3sRT2h1"]
+  Port = 3919
+  # Uncomment to override the node's default addresses announced in the network
+  AnnouncedAddresses = ["/dns4/example.com/tcp/3919", "/ip4/80.70.60.50/tcp/3919"]
+
+[TSS]
+# Timeout for TSS protocol pre-parameters generation. The value
+# should be provided based on resources available on the machine running the client.
+# This is an optional parameter, if not provided timeout for TSS protocol
+# pre-parameters generation will be set to `2 minutes`.
+  PreParamsGenerationTimeout = "2m30s"
+
+```
+
+==== Parameters
```

Comment:
I tried to think about a better table layout for these parameters.  The non-uniform size of each of them is hard to read.  Might want to make it 5 columns wide and just include the left-most column header currently an entire new column all the way down.  Then you'd be uniform.  

Something like this? 

[%header,cols=5*]
|===
|Type
|Description
|Default
|Required

|`ethereum`
|`URL`
|The Ethereum host your keep-ecdsa will connect to.  Websocket protocol/port.
|""
|Yes

Reply:
Sure, can see what it looks like.

Reply:
I had a look at how this renders on our docs page: https://docs.keep.network/run-random-beacon.html#_application, and they line up.  I guess this is a product of how GH renders adoc files.

Reply:
Noting ^ is a different document, but formatted the same way.